src/Storage/Configuration/Providers.cs

```diff
@@ -21,18 +31,178 @@ public class Providers : SnConfig
             typeof(ContentRepository.Storage.Security.Sha256PasswordHashProviderWithoutSalt).FullName);
         public static string SkinManagerClassName { get; internal set; } = GetProvider("SkinManager", "SenseNet.Portal.SkinManager");
         public static string DirectoryProviderClassName { get; internal set; } = GetProvider("DirectoryProvider");
+        public static string SecurityDataProviderClassName { get; internal set; } = GetProvider("SecurityDataProvider",
+            typeof(EF6SecurityDataProvider).FullName);
         public static string SecurityMessageProviderClassName { get; internal set; } = GetProvider("SecurityMessageProvider", 
             typeof(DefaultMessageProvider).FullName);
         public static string DocumentPreviewProviderClassName { get; internal set; } = GetProvider("DocumentPreviewProvider",
             "SenseNet.Preview.DefaultDocumentPreviewProvider");
         public static string ClusterChannelProviderClassName { get; internal set; } = GetProvider("ClusterChannelProvider",
             typeof(VoidChannel).FullName);
 
+        public static string ElevatedModificationVisibilityRuleProviderName { get; internal set; } =
+            GetProvider("ElevatedModificationVisibilityRuleProvider",
+                "SenseNet.ContentRepository.SnElevatedModificationVisibilityRule");
+
         public static bool RepositoryPathProviderEnabled { get; internal set; } = GetValue<bool>(SectionName, "RepositoryPathProviderEnabled", true);
 
         private static string GetProvider(string key, string defaultValue = null)
         {
             return GetString(SectionName, key, defaultValue);
         }
+
+        //===================================================================================== Instance
+
+        /// <summary>
+        /// Lets you access the replaceable providers in the system. This instance may be replaced 
+        /// by a derived special implementation that stores instances on a thread context 
+        /// for testing purposes.
+        /// </summary>
+        public static Providers Instance { get; set; } = new Providers();
+
+        //===================================================================================== Named providers
+

```

Comment:
Use #region for each well known items e.g. private Lazy<DataProvider> _dataProvider = new Lazy<DataProvider>(() =>